tests/pytests/test_profile.py

```diff
@@ -807,3 +810,42 @@ def verify_search_mode(query_type, query, params, expected_mode, expected_iterat
   # High estimated results → starts BATCHES, but 0 actual results → switches to ADHOC_BF
   verify_search_mode('SEARCH', '(@t:hello other)=>[KNN 3 @v $vec BATCH_SIZE 100]', ['vec', '????????'], 'HYBRID_BATCHES_TO_ADHOC_BF')
   verify_search_mode('AGGREGATE', '(@t:hello other)=>[KNN 3 @v $vec BATCH_SIZE 100]', ['vec', '????????'], 'HYBRID_BATCHES_TO_ADHOC_BF')
+
+def testPofileGILTime():
+  env = Env(moduleArgs='WORKERS 1')
+  conn = getConnectionByEnv(env)
+
+  # Populate db
+  with env.getClusterConnectionIfNeeded() as conn:
+    for i in range(100):
+      res = conn.execute_command('hset', f'doc{i}',
+                      'f', 'hello world',
+                      'g', 'foo bar',
+                      'h', 'baz qux')
+
+  env.cmd('ft.create', 'idx', 'SCHEMA', 'f', 'TEXT', 'g', 'TEXT', 'h', 'TEXT')
+  res = env.cmd('FT.PROFILE', 'idx', 'AGGREGATE', 'query', 'hello' ,'SORTBY', '1', '@f')
+
+  # Record structure:
+  # ['Type', 'Threadsafe-Loader', 'GIL-Time', ANY , 'Time', ANY, 'Counter', 100]
+  # ['Total GIL time', ANY]
+
+  try:
+    # env.assertTrue(recursive_contains(res, 'Threadsafe-Loader'), message=f"res: {res}")
+    # env.assertTrue(recursive_contains(res, 'Total GIL time'), message=f"res: {res}")
+
+    # extract the GIL time of the threadsafe loader result processor
+    rp_index = recursive_index(res, 'Threadsafe-Loader')[:-1]
+    rp_record = access_nested_list(res, rp_index)
+    rp_GIL_time = rp_record[rp_record.index('GIL-Time') + 1]
+
+    # extract the total GIL time
+    total_GIL_index = recursive_index(res, 'Total GIL time')
+    total_GIL_index[-1] += 1
+    total_GIL_time = access_nested_list(res, total_GIL_index)
+
+    env.assertGreaterEqual(float(total_GIL_time), 0)
+    env.assertGreaterEqual(float(rp_GIL_time), 0)
+    env.assertGreaterEqual(float(total_GIL_time), float(rp_GIL_time))
```

Comment:
@Itzikvaknin what is the purpose of this assert? `assertGreaterEqual` will always be true isn't it? Is there a way to test that we waited for the GIL in a non-flaky way?